lib/parallel_tests.rb

```diff
@@ -20,12 +20,27 @@ def determine_number_of_processes(count)
       ].detect{|c| not c.to_s.strip.empty? }.to_i
     end
 
+    def with_pid_file
+      Tempfile.open('parallel_tests-pidfile') do |f|
+        ENV['PARALLEL_PID_FILE'] = f.path
+        # Pids object should be created before threads will start adding pids to it
+        # Otherwise we would have to use Mutex to prevent creation of several instances
+        @pids = pids
```

Comment:
if `pids` raises then the ENV would not be unset, so it should be inside the begin ... I can fix that on master